### PR TITLE
monitoring: push build stats to database

### PR DESCRIPTION
**Pull request by sodar**

### CODE REVIEW

File: monitoring/collect_and_push_job_stats.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+import argparse
+import getpass
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from stats import collect_and_push_job_stats
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
```
Comment (jablonskim) [inline monitoring/collect_and_push_job_stats.py]: Maybe create some `MysqlCommonArgumentParser` for all `--mysql-*` arguments. See an example [here](https://github.com/codilime/contrail-windows-ci/blob/development/ansible/files/vmware_api.py#L16).

File: monitoring/collect_and_push_job_stats.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    args = parser.parse_args()
+
+    if not args.mysql_password:
+        prompt = 'Enter password (for MySQL user {}): '.format(args.mysql_username)
+        args.password = getpass.getpass(prompt=prompt)
+
+    return args
+
+
+def get_mysql_connection_string(host, username, password, database):
```
Comment (jablonskim) [inline monitoring/collect_and_push_job_stats.py]: Please move this function to some `common` file. Another good idea is to make it a method of `MysqlCommonArgumentParser` class.

**Review by jablonskim — CHANGES_REQUESTED** [review]

File: monitoring/database.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class Build(MonitoringBase):
+    __tablename__ = 'builds'
+
+    id = Column(Integer, primary_key=True)
+    job_name = Column(String(4096), nullable=False)
+    build_id = Column(Integer, nullable=False)
+    build_url = Column(String(4096), nullable=False)
+    finished_at = Column(BigInteger, nullable=False)
+    status = Column(String(4096), nullable=False)
+    duration = Column(BigInteger, nullable=False)
```
Comment (krdln) [inline monitoring/database.py]: Let's include `secs` / `millis` in column names `finished_at` and `duration` as we've discussed yesterday.

**Review by krdln — CHANGES_REQUESTED** [review]

File: monitoring/collect_and_push_job_stats.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
```
Comment (krdln) [inline monitoring/collect_and_push_job_stats.py]: Shouldn't it be `python3`? Also, do you think it's a good idea to `chmod +x` these files?

File: monitoring/test_stats.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
+import unittest
```
Comment (krdln) [inline monitoring/test_stats.py:2]: Maybe add `#!/usr/bin/env python3` here too?

File: monitoring/collect_and_push_build_stats.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+from common import get_mysql_connection_string, MysqlCommonArgumentParser
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from stats import collect_and_push_job_stats
```
Comment (jablonskim) [inline monitoring/collect_and_push_build_stats.py]: Should be `collect_and_push_build_stats`

**Review by jablonskim — CHANGES_REQUESTED** [review]

File: monitoring/provision.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def provision_database(connection_string, model):
+    engine = create_engine(connection_string, echo=True)
+    model.metadata.create_all(engine)
+
+
+def main():
+    args = parse_args()
+    connection_string = get_mysql_connection_string(host=args.host, username=args.username,
```
Comment (jablonskim) [inline monitoring/provision.py]: Should be `args.mysql_host`, `args.mysql_username`, ...

File: monitoring/provision.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+import argparse
+import getpass
```
Comment (jablonskim) [inline monitoring/provision.py]: We do not need `argparse` and `getpass` here anymore

**Review by jablonskim — CHANGES_REQUESTED** [review]

**Comment by krdln** [comment]
Should we add github- and gerrit-related params to the database (such as `ZUUL_CHANGE`, `ZUUL_PATCHSET`, `ghprbActualCommit` and `ghprbPullId`)? (So we can analyse rechecks and provide links to gerrit/gh)

**Comment by sodar** [comment]
I think we can add it in later PRs, possibly using some database migrations: https://pypi.python.org/pypi/alembic

**Review by krdln — DISMISSED** [review]

**Review by jablonskim — APPROVED** [review]

**Review by krdln — APPROVED** [review]

**Pull request merged**